Refactor Table of Contents for Integrate section

neflyte opened the pull request:
The Table of Contents of the Integrate section has been refactored to better organize and surface information. Changes consist mainly of page relocations and link updates.

The “Development Guides” card on the docs homepage was removed as it had significant overlap with the “Integrate” card.

In addition to the docs update, a `Makefile` target for building the docs locally without re-fetching manifest reference information was added.

mattermod commented:
Hello @neflyte,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

neflyte commented:
/update-branch

mattermod commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

azigler commented:
/update-branch

mattermod commented:
We don't have permissions to update this PR, please contact the submitter to apply the update.

neflyte commented:
/update-branch

pfltdv commented:
/update-branch

wiersgallak commented:
One thing that I notice right away is that the Getting Started Page does not mirror the LHS navigation in the Integrate & Extend section. On the Getting Started, I also think we should make Slash Commands, Webhooks, Embed, Customize, and Interactive Messages as equals in weight to the others. As for the order, let's start with simple to advanced and maybe even signify this in the content at the start of the page.

wiersgallak reviewed (commented):
I didn't review all files in detail, but left general comment about Integrate & Extend LHS navigation.

cwarnermm commented:
/update-branch

cwarnermm commented:
@neflyte - This is a great starting point! Two small pieces of feedback:
- On the landing page, let's update the Integrate & Extend card description to list options in order of simplest to most complex
- The Mattermost Dev Talks page takes users to a YouTube channel with content from at least a year ago. Proposing that this section be removed from the docs altogether. @azigler Thoughts?

cwarnermm commented:
Reviewers: Please note that this PR focuses on re-organizing existing content. Once the information architecture and site organization is updated, focus will turn to content-level updates.

azigler reviewed (commented):
@neflyte Great work here with this PR. Overall, I think we're heading in the right direction. Here's my feedback:

- You've done a great job bundling together relevant documentation, but I think it points out a problem with our LHS menu for this new organized approach. Since everything in the LHS TOC starts out closed, in order to find nested links you have to click a few times to expand everything. It turns finding documentation into a hunt. Is it possible for us to move to a system where everything is collapsed open by default, or get rid of the collapsing altogether? CC @cwarnermm 
- I noticed that the FAQ section has a [+] collapse button but nothing nested underneath it.
- https://developers.mattermost.com/ going from 3 columns to 2 columns makes the remaining text stretch and wrap in an unappealing way. Whether now or later, I think we should improve that copy to better fit its new space:
<img width="1362" alt="image" src="https://user-images.githubusercontent.com/7295363/183524155-4b54e63a-8f61-478e-aef7-055e912cba0e.png">

- Related, we'll need to make sure the header gets updated in tandem with this change:
<img width="829" alt="image" src="https://user-images.githubusercontent.com/7295363/183523949-539bbd5d-c88b-4e34-ae08-af3899f9ef2b.png">

cwarnermm commented:
I agree with @azigler that a collapsed-by-default LHS requires users to explore to find the information they need (particularly when the site doesn't yet have a search option). However, this is the same user experience provided in the product documentation and the API reference currently, so there's consistency. 

@neflyte has a card on his board to implement search; we're currently holding off on that until more of the content rework is in place.

The mis-match between the header and the new dev docs landing page is a good point; however, those sections don't match in production now either (the Deploy section takes users to https://mattermost.com/deploy/).

neflyte commented:
@azigler Thanks for the review!
I've fixed the issue where the FAQ section has a `[+]` and should not, and I've also fixed the issue where the `/integrate/apps/using-mattermost-api/` path points to an outdated page.

I can make the 2 cards on the home page have the same size as the 3 cards used to have, if that helps.

Regarding the site header links, there wasn't a `Deploy` section in the dev docs so I'm not sure where that link is supposed to point. I can rearrange the main links in the header so they match what's on the home page as well. (e.g. Contribute, followed by Integrate)

Thoughts?

neflyte commented:
I went ahead and made the 2 cards have the same size as the previous 3-card homepage did, and I've updated the text on the Integrate & Extend card to list technologies in ascending order of complexity.

neflyte commented:
I have also fixed the issue where a ToC section would not expand to show child documents if the section itself was clicked-on. Child pages of a ToC section will now expand when the reader is anywhere in the section, including the section "homepage" itself.

Note that the current implementation of the ToC is limited to 2 levels deep of content. It may be possible to render all levels of content, however. I have a card on my board to investigate it.

azigler commented:
> @azigler Thanks for the review! I've fixed the issue where the FAQ section has a `[+]` and should not, and I've also fixed the issue where the `/integrate/apps/using-mattermost-api/` path points to an outdated page.
> 
> I can make the 2 cards on the home page have the same size as the 3 cards used to have, if that helps.
> 
> Regarding the site header links, there wasn't a `Deploy` section in the dev docs so I'm not sure where that link is supposed to point. I can rearrange the main links in the header so they match what's on the home page as well. (e.g. Contribute, followed by Integrate)
> 
> Thoughts?

I agree reverting the 2 cards to be their old size was the best move, that looks a lot better. As for the site header, we would need to get @emdecr involved so we can all stay on the same page. They are planning changes on their side, and frankly I'm more apt to go with their arrangement suggestion because they have the harder job (reshuffling that header is a big task), and we can just follow suit and update our dev docs landing page to reflect their best iteration. CC'ing @cwarnermm on this idea.

emdecr commented:
Thanks for the flag re: header/nav y'all! The header's gone through a few iterations 😅 And right now the nav on mm.com is getting a revamp as we speak. Once that is nailed down, I can circle-back to dev.mm.com and docs.mm.com to sync the header across all sites. Does that work for folks?

neflyte commented:
I think that's the approach we should take, @emdecr. :smile: It would allow this PR (and #1099) to move forward.

cwarnermm reviewed (approved):
Thank you for this great PR, @neflyte!

azigler reviewed (approved):
Great work here @neflyte!

mickmister commented on site/content/integrate/apps/drivers/nodejs.md:
```diff
@@ -0,0 +1,8 @@
+---
+
+### Coming Soon!
+In the future this page will describe how to use the NodeJS driver in your App.
```
@neflyte I'm thinking we should remove these pages until we have the drivers. It's confusing to have it a promoted page in the left sidebar, and then have no content here.
